core/src/main/scala/org/apache/spark/ui/exec/ExecutorsPage.scala

```diff
@@ -53,6 +54,9 @@ private[ui] class ExecutorsPage(
   // When GCTimePercent is edited change ToolTips.TASK_TIME to match
   private val GCTimePercent = 0.1
 
+  // a safe String to Int for sorting ids (converts non-numeric Strings to -1)
+  private def strToInt(str: String) : Int = Try(str.toInt).getOrElse(-1)
```

Comment:
That was my intention and why I chose to leave this function private in ExecutorsPage rather than in a Utils file